sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -889,6 +889,9 @@ object SparkSession extends api.BaseSparkSessionCompanion with Logging {
 
         // No active nor global default session. Create a new one.
         val sparkContext = userSuppliedContext.getOrElse {
+          // Override appName with the submitted appName
+          sparkConf.getOption("spark.submit.appName")
```

Comment:
I feel like this is a bit arguable because it's going to overwrite the configurations already specified by users.

Reply:
Could you elaborate what kind of confirmation are overwritten here, @HyukjinKwon ?
> I feel like this is a bit arguable because it's going to overwrite the configurations already specified by users.

Reply:
Oh I think I missed the discussion above. So this one is sort of a special conf. LGTM then.

Reply:
Thank you, Hyukjin! For the Python part, I'll make a new PR by addressing your comment in a few days.